### PR TITLE
German translation fixes in security form

### PR DESCRIPTION
**Pull request by saschaszott**
## Description
Fixed some German translation in `de.json5`.

## Instructions for Reviewers
List of changes in this PR:
* some translation keys were **added**
* some obsolete translation keys which are not present in `en.json5` were **removed**

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [ ] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [ ] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [ ] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by saschaszott** [comment]
Additionally, the default minimum password length was changed to 8 characters with commit https://github.com/DSpace/DSpace/commit/a31bb4bf6198fa6abaa918edd9368a7fb43f75b7. We should better remove the length value in translated messages.

File: src/assets/i18n/de.json5 [hunk 1]
```diff
@@ -3838,17 +3838,17 @@
...
   
   // "profile.security.form.label.password": "Password",
   "profile.security.form.label.password": "Passwort",
   
   // "profile.security.form.label.passwordrepeat": "Retype to confirm",
-  "profile.security.form.label.passwordrepeat": "Zum Bestätigen erneut eingeben",
+  "profile.security.form.label.passwordrepeat": "Zum Bestätigen das Passwort erneut eingeben",
+  
+  // "profile.security.form.label.current-password": "Current password"
+  "profile.security.form.label.current-password": "Ihr aktuelles Passwort"
```
Comment (tdonohue) [inline src/assets/i18n/de.json5:3851]: @saschaszott : You are missing a comma after this line, which is causing test failures.  This file isn't valid JSON right now.

Comment (saschaszott) [reply src/assets/i18n/de.json5]: Sorry for that and thank you for reviewing my changes!

Comment (saschaszott) [reply src/assets/i18n/de.json5]: I've added the missing comma.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @saschaszott !  Looks good now, so I'm merging immediately.

**Pull request merged**